### DISCUSSION TITLE
Play tracking in spotify

### DISCUSSION THREAD

**Discussion by marcohamersma** [music-assistant/support#2664] [Feature requests and ideas]
When I play tracks on Spotify through Music Assistant, it doesn't seem to report the tracks to Spotify as played (I checked through the API). Do excuse me if it's an issue with my setup, but otherwise I hope you might be able to implement this feature, as it's important for recommendations and the personalised playlists.

My needs might also be covered by #632, but regardless; reporting plays back to the source service seems like it should be default behaviour (also for other sources). Would you consider adding this feature?

**Comment by OzGav** [comment]
The complexity is that MA users can have multiple Spotify accounts and multiple providers. So if you play a playlist for example should all Spotify accounts have this information passed to them and should all other providers get that info as well? Similarly, if you listen to a local track should that be reported to Spotify?

**Reply by marcohamersma** [reply]
For me this is about covering the expected behavior (tracking the plays from tracks in someone's music library) in the most common use cases.

I'm assuming here that it's the most common usecase to have only one account for music streaming services connected, so I'd suggest the reporting happens to the source account of the track or the playlist that's being played. 

I realize this is more complex if multiple sources/accounts have the same track. But that doesn't seem to be the most common case, so whichever solution is used there (not tracking/sending to the first/sending to all involved sources) is not _that_ important to me. I'd leave that to your insight.

Local tracks would not have a source service so they wouldn't be covered by play reporting (unless [scrobbling](https://github.com/orgs/music-assistant/discussions/632), but that is an additional issue as it doesn't solve the recommendation issues).

I'd be happy to help explore how it could be done.

**Reply by domemshur** [reply]
I agree. It needs to track info or it's unusable

**Comment by Kilkenny-82** [comment]
It seems this conversation has gone silent quick and can't find anything else on this idea. I have added my Jellyfin and Navidrome (SubSonic) hosts to it, but also for these no plays are recorded to the host. Any news on this?

**Reply by OzGav** [reply]
Has your question got anything to do with Spotify?

**Reply by Kilkenny-82** [reply]
My comment was more saying it's not only Spotify, this feature would also be nice to have on other hosted environments like Jellyfin or SubSonic

**Reply by OzGav** [reply]
Currently there are no plans to scrobble to the individual providers. Indeed it was just removed from the subsonic provider as it was unsupportable. Lastfm, LibreFm,listenbrainz have been added as plugins.

**Reply by Kilkenny-82** [reply]
Pity, though understandable it would bring a big level of complexity to the project

**Reply by domemshur** [reply]
> Currently there are no plans to scrobble to the individual providers. Indeed it was just removed from the subsonic provider as it was unsupportable. Lastfm, LibreFm,listenbrainz have been added as plugins.

Are there any plans on making playback tracking available? I like music assistant, but i cannot use it until it can track playback history. Thanks.

**Reply by OzGav** [reply]
There is no Spotify API for this so if Spotify doesn’t track it as tracks are played then there is no way to do it. MA supports other scrobblers and tracks playback history internally

**Reply by domemshur** [reply]
How is it possible to track it through Spotify connect then? That's what I use to track playback

**Reply by OzGav** [reply]
Sorry don't know. I just use MA for everything.

**Comment by hazelk500** [comment]
Good point—play tracking is essential for accurate recommendations and personalized playlists. Reporting plays back to Spotify (and other sources) would definitely make Music Assistant feel more complete and consistent.

**Comment by domemshur** [comment]
I don't use music Assistant at the moment because of this. I have to do some weird Spotify connect loophole. I am hoping they can get this figured out soon
